zephyr/include/sof/lib/regions_mm.h

```diff
@@ -10,31 +10,91 @@
 
 #include <adsp_memory.h>
 #include <adsp_memory_regions.h>
+#include <ipc/topology.h>
+#include <rtos/alloc.h>
 #include <zephyr/drivers/mm/system_mm.h>
 #include <zephyr/sys/mem_blocks.h>
 #include <zephyr/init.h>
-#include <ipc/topology.h>
 
-/*
- * Struct containing information on virtual memory heap.
- * Information about allocated physical blocks is stored in
- * physical_blocks_allocators variable and uses zephyr memblocks api.
- */
+#define MAXIMUM_BLOCK_SIZES 8
+
+//list of virtual_memory_heap
+sys_slist_t vm_heaps;
+sys_slist_t physical_mappings;
+
+
 struct virtual_memory_heap {
+	sys_snode_t node;
 	/* zephyr provided virtual region */
-	struct sys_mm_drv_region *virtual_region;
-	/* physical pages allocators represented in memblocks */
-	struct sys_multi_mem_blocks physical_blocks_allocators;
-	/* SOF memory capability */
-	uint32_t memory_caps;
+	struct sys_mm_drv_region* virtual_region;
+	/* physical pages allocators represented in memblocks memblocks sizes are passed on heap creation */
+	struct sys_mem_blocks *physical_blocks_allocators[MAXIMUM_BLOCK_SIZES];
+	
+	//keep current active allocations - further down the line it can be transitioned to memblocks bitarray
+	sys_slist_t active_allocations;
+	//allocations will be contingous
+	bool span;
 };
 
-/* Available externaly array containing all information on virtual heaps
- * Used to control physical allocations and overall virtual to physicall
- * mapping on sof side (zephyr handles the actual assigning physical memory
- * sof only requests it).
- */
-extern struct virtual_memory_heap
-	vm_heaps[CONFIG_MP_MAX_NUM_CPUS + VIRTUAL_REGION_COUNT];
+struct active_physical_mapping
+{
+	sys_snode_t node;
+
+	uint32_t *mapping_ptr;
```

Comment:
Is this a pointer to a 32-bit variable or just a memory address (should be void * then)?